docs/examples/POWGEN_data_reduction.ipynb

```diff
@@ -0,0 +1,424 @@
+    "    ValidTofRange: sc.array(dims=['tof'], values=[0.0, 16666.67], unit='us'),\n",
+    "\n",
+    "    # Edges for binning in d-spacing\n",
+    "    DspacingBins: sc.linspace('dspacing', 0.0, 2.3434, 200, unit='angstrom'),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3d9b9bd-6923-4124-954f-e92672845be2",
+   "metadata": {},
+   "source": [
+    "## Create pipeline using Sciline\n",
+    "\n",
+    "We use the basic providers available in `essdiffraction` as well as the specialised `powder` and `powgen` providers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "35412429-c039-4591-b8b8-49b11a128b7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "providers = [\n",
+    "    *ess.diffraction.providers,\n",
+    "    *powder.providers,\n",
+    "    *powgen.providers,\n",
+    "]\n",
+    "pipeline = sciline.Pipeline(\n",
+    "    providers,\n",
+    "    params=params,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4af9055-385d-41a4-9f1c-35fd1e232565",
+   "metadata": {},
+   "source": [
+    "## Use the pipeline\n",
+    "\n",
+    "### Compute final result\n",
+    "\n",
+    "We can get the graph for computing the final d-spacing histogram:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d09bc8d5-4251-483f-bb45-aa8cf67f69b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dspacing_histogram = pipeline.get(DspacingHistogram)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "67288e4a-5140-430d-a15a-7caa0f365904",
+   "metadata": {},
+   "source": [
+    "Before computing the result, we visualize the pipeline:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "11fea84f-aef5-4d31-94c4-1f710c7f8836",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# left-right layout works better for this graph\n",
+    "dspacing_histogram.visualize(graph_attr={'rankdir': 'LR'})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a3177aeb-51ae-4166-907b-1caa9cf751aa",
+   "metadata": {},
+   "source": [
+    "Now we compute the result:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "14663b68-c9b8-444a-bf06-7454b67f8d82",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result = dspacing_histogram.compute()\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94792ed7-cfba-4467-9b5e-91211c5f2d89",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6335d026-d526-4ad4-bcda-7270e955d2c1",
+   "metadata": {},
+   "source": [
+    "### Save reduced data to file\n",
+    "\n",
+    "We ultimately need to write the reduced data to a file.\n",
+    "This could be done with the `result` we computed above.\n",
+    "But then we would have to provide all metadata manually.\n",
```

Comment:
I suppose there is no metadata here? This is a bit confusing maybe?

Reply:
The filename, sort of. But yes, there isn't any real metadata. I can change it now and revert later when we write a cif file instead.